server/src/main/java/org/apache/druid/discovery/BaseNodeRoleWatcher.java

```diff
@@ -264,8 +267,15 @@ private void cacheInitializedTimedOut()
   @GuardedBy("lock")
   private void cacheInitialized(boolean timedOut)
   {
+    // This method is called only once with either timedOut = true or false, but not both.
```

Comment:
nit: usually such comments are meant to be placed above the method and not inside.

Reply:
done